### DISCUSSION TITLE
Rewriting the RSS links

### DISCUSSION THREAD

**Discussion by anthonydresser** [martinrotter/rssguard#1304] [Q&A]
I want to rewrite the links from an RSS feed since they are relative links rather than absolute, so RSS guard can't open them correctly. I assume I can do this using the post processing script, but I couldn't find any documentation on what the expectations of this feature are (what input variables are available, what the return expectations are etc.)

**Comment by martinrotter** [comment]
Give link to feed first, this might be a bug

**Reply by anthonydresser** [reply]
It is a private rss feed, so unfortunately I can't share the feed. Because it is private and I need to provide creds, I am also using a curl command to fetch the feed so I'm not sure if that would affect things.

**Reply by martinrotter** [reply]
I see. Usually, RSS Guard does support relative links and resolves those automatically to absolute links. That's why I asked about it.

If you use curl command to get it instead of regular URL in RSS Guard then that is the reason the built-in automatic relative URL resolution fails.

To fix this, you can use simple article filter similar to this:

```js
function filterMessage() {
  msg.url = "http://myabsolutewebsite.com" + msg.url;
  return MessageObject.Accept;
}
```

THis filter prepends URL of article with custom absolute URL. Maybe you could post excerpt of feed contents to see how exactly those relative URLs look in feed data. I guess they are starting with "/"? If so the above filter should work.

Make sure to read the related docs.

https://rssguard.readthedocs.io/en/stable/features/filters.html#article-filters-dialog

In this case, using post-process script to fixup those URLs is also possible but is likely not necessary, the above article filter will do the job.
